src/gui/sharemanager.cpp

```diff
@@ -255,10 +256,20 @@ QString LinkShare::getToken() const
 
 void LinkShare::setExpireDate(const QDate &date)
 {
+    shareJob(&LinkShare::slotExpireDateSet)->setExpireDate(getId(), date);
+}
+
+void LinkShare::setLabel(const QString &label)
+{
+    shareJob(&LinkShare::slotLabelSet)->setLabel(getId(), label);
+}
+
+template <typename LinkShareSlot>
+OcsShareJob *LinkShare::shareJob(const LinkShareSlot slot) {
```

Comment:
Could maybe be confusing if we use the keyword `slot` here as a variable specifier.